server/instance.go

```diff
@@ -149,6 +150,25 @@ func (s *Server) registerInstanceRoutes(g *echo.Group) {
 					return err
 				}
 			}
+			// Ensure all databases belong to this instance are under the default project before instance is archived.
+			if v := instancePatch.RowStatus; v != nil && *v == api.Archived.String() {
+				databases, err := s.store.FindDatabase(ctx, &api.DatabaseFind{InstanceID: &id})
+				if err != nil {
+					return echo.NewHTTPError(http.StatusInternalServerError,
+						fmt.Errorf("Failed to find databases in the instance %d", id)).SetInternal(err)
+				}
+				var databaseNameList []string
+				for _, database := range databases {
+					if database.ProjectID != api.DefaultProjectID {
+						databaseNameList = append(databaseNameList, database.Name)
+					}
+				}
+				if len(databaseNameList) > 0 {
+					return echo.NewHTTPError(http.StatusBadRequest,
+						"You should transfer these databases to the default project before archiving the instance: "+
```

Comment:
Can we use fmt.Sprintf() which is easier to understand?

Reply:
done!